src/Jackett.Common/Indexers/FileListApi.cs

```diff
@@ -0,0 +1,204 @@
+        {
+            LoadValuesFromJson(configJson);
+            logger.Info("Testing provider filelist...");
+            var pingResponse = await CallProviderAsync(new TorznabQuery());
+
+            try
+            {
+                var json = JArray.Parse(pingResponse);
+                if (json.Count > 0)
+                {
+                    IsConfigured = true;
+                    SaveConfig();
+                    return IndexerConfigurationStatus.Completed;
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new ExceptionWithConfigData(ex.Message, configData);
+            }
+            return IndexerConfigurationStatus.RequiresTesting;
+
+        }
+
+        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        {
+            var releases = new List<ReleaseInfo>();
+            var response = await CallProviderAsync(query);
+
+            if (response.StartsWith("{\"error\""))
+                throw new ExceptionWithConfigData(response, configData);
+
+            try
+            {
+                var json = JArray.Parse(response);
+                foreach (var row in json)
+                {
+                    var detailsUri = new Uri(DetailsUrl + "?id=" + (string)row["id"]);
+                    var seeders = (int)row["seeders"];
+                    var peers = seeders + (int)row["leechers"];
+                    var publishDate = DateTimeUtil.FromFuzzyTime((string)row["upload_date"] + " +0200");
+                    var downloadVolumeFactor = (int)row["freeleech"] == 1 ? 0 : 1;
+                    var release = new ReleaseInfo
+                    {
+                        Title = (string)row["name"],
+                        Comments = detailsUri,
+                        Link = new Uri((string)row["download_link"]),
+                        Category = MapTrackerCatDescToNewznab((string)row["category"]),
+                        Size = (long)row["size"],
+                        Files = (long)row["files"],
+                        Grabs = (long)row["times_completed"],
+                        Seeders = seeders,
+                        Peers = peers,
+                        MinimumRatio = 1,
+                        MinimumSeedTime = 172800,
+                        PublishDate = publishDate,
+                        UploadVolumeFactor = 1,
+                        DownloadVolumeFactor = downloadVolumeFactor,
+                        Guid = detailsUri
+
+                    };
+                    release.Imdb = ((JObject)row).ContainsKey("imdb") ? ParseUtil.GetImdbID((string)row["imdb"]) : null;
+
+                    releases.Add(release);
+                }
+                return releases;
+            }
+            catch (Exception ex)
+            {
+                OnParseError(response, ex);
+            }
+            return null;
```

Comment:
PerformQuery should not return null. Current practice is
```suggestion
            return releases;
```